src/overlays/actors/ovl_Bg_Kin2_Picture/z_bg_kin2_picture.c

```diff
@@ -34,73 +39,306 @@ const ActorInit Bg_Kin2_Picture_InitVars = {
+static ColliderTrisInit sTrisInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_TRIS,
+    },
+    ARRAY_COUNT(sTrisElementsInit),
+    sTrisElementsInit,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B6FA24[] = {
-    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_CONTINUE),
-    ICHAIN_F32_DIV1000(terminalVelocity, -20000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneScale, 100, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_CONTINUE),
-    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+static Vec3f sDustBasePos = { 0.0f, 23.0f, 0.0f };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_CONTINUE),  ICHAIN_F32_DIV1000(terminalVelocity, -20000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_CONTINUE), ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+s32 BgKin2Picture_IsSkulltulaCollected(PlayState* play, s32 skulltulaParams) {
+    s32 flag = -1;
+
+    if ((u8)skulltulaParams & 3) {
+        flag = BG_KIN2_PICTURE_GETS_3FC(skulltulaParams);
+    }
+
+    return (flag >= 0) && Flags_GetTreasure(play, flag);
+}
+
+void BgKin2Picture_SpawnSkulltula(BgKin2Picture* this, PlayState* play2) {
+    PlayState* play = play2;
+    s32 actorSpawnParams;
+
+    if (!BG_KIN2_PICTURE_GET_PARAMS_05(&this->dyna.actor)) { // Gold Skulltula is still here.
+        actorSpawnParams = BG_KIN2_PICTURE_SKULLTULA_SPAWN_PARAM(&this->dyna.actor);
+        if (!BgKin2Picture_IsSkulltulaCollected(play, actorSpawnParams) &&
+            Actor_Spawn(&play->actorCtx, play, ACTOR_EN_SW, this->dyna.actor.home.pos.x,
+                        this->dyna.actor.home.pos.y + 23.0f, this->dyna.actor.home.pos.z, 0,
+                        this->dyna.actor.home.rot.y, 0, actorSpawnParams)) {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+    }
+}
+
+void BgKin2Picture_SpawnDust(BgKin2Picture* this, PlayState* play) {
+    f32 temp_fs0;
+    Vec3f basePos;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    s32 temp_s1;
+    s32 temp2;
+    s32 scale;
+    s16 scaleStep;
+    s32 phi_s3;
+    s32 i;
+
+    Matrix_SetTranslateRotateYXZ(this->dyna.actor.world.pos.x,
+                                 this->dyna.actor.world.pos.y +
+                                     (this->dyna.actor.shape.yOffset * this->dyna.actor.scale.y),
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+    Matrix_MultVec3f(&sDustBasePos, &basePos);
+    pos.y = basePos.y - 7.0f;
+    velocity.y = 0.0f;
+    accel.y = 0.2f;
+
+    for (i = 0, phi_s3 = 0; i < 20; i++, phi_s3 += 0xCCC) {
+        temp2 = Rand_ZeroOne() * 3276.0f;
+        temp_s1 = temp2 + phi_s3;
```

Comment:
```suggestion
        temp_s1 = (s32)(Rand_ZeroOne() * 3276.0f) + phi_s3;
```
Is temp2 needed? Does this work by chance?